app/controllers/application_controller.rb

```diff
@@ -367,6 +367,7 @@ def report_data
     if options.nil? || options[:view].nil?
       model_view = process_params_model_view(params, options)
       @edit = session[:edit]
+      options_from_session(options)
```

Comment:
> Generally I consider the idea of "Fixed `report_data` to take options(set in get_view) into account" a step back.

Ok, I need you to re-read the words "Fixed report_data" :) - meaning, `report_data` has a bug that has been fixed with that commit https://github.com/ManageIQ/manageiq-ui-classic/pull/1553/commits/5e78af22f39cb97b6017d8abdb1a8db03187f086
As a matter of fact, I think I have found a couple of issues with `report_data`... (which we can discuss later)

One thing that surprised me here as I was debugging `get_view` and `report_data` was the fact that we call `get_view` twice!! ... From `nested_list` and then from `report_data`. Why?
I mean, it's executing the same queries at least twice. Now that is what I call a step backward.

Not to mention the fact that `report_data` completely ignores my `where_clause` :(

Good news is,  https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/controllers/application_controller.rb#L1512-L1528 stores the `where_clause` in the `session`.
Bad news is, I have to use `session` to retrieve it.
I have my hangups about using `session` as much as you do.